### PR TITLE
Typescript with flycheck

### PR DESCRIPTION
**Pull request by hamhut1066**
this aims to replace the use of flymake with flycheck, giving us a consistent fly* package.

### CODE REVIEW

**Comment by robbyoconnor** [comment]
Rebase so it's one single commit ;)

File: contrib/!lang/typescript/packages.el [hunk 1]
```diff
@@ -4,20 +4,62 @@
...
 ;; Author: Chris Bowdon <c.bowdon@bath.edu>
+;; Author: Hamish Hutchings <hamhut1066@gmail.com>
 ;; URL: https://github.com/syl20bnr/spacemacs
 ;;
 ;; This file is not part of GNU Emacs.
 ;;
 ;;; License: GPLv3
 
-(setq typescript-packages '(tss))
+(setq typescript-private-packages '(
```
Comment (person808) [inline contrib/!lang/typescript/packages.el]: `typescript-packages`

File: contrib/!lang/typescript/packages.el [hunk 2]
```diff
@@ -4,20 +4,62 @@
...
+                                    tss
+                                    flycheck))
 
-(defun typescript/init-tss ()
-  "Initialize my package"
+(setq test-layer-excluded-packages '(flymake))
```
Comment (person808) [inline contrib/!lang/typescript/packages.el]: Is this necessary? `Flymake` is built in and doesn't interfere if its not activated

File: contrib/!lang/typescript/packages.el [hunk 3]
```diff
@@ -4,20 +4,62 @@
...
+
+(defun typescript-private/post-init-flycheck ()
```
Comment (person808) [inline contrib/!lang/typescript/packages.el]: `typescript/post-init-flycheck`

File: contrib/!lang/typescript/packages.el [hunk 4]
```diff
@@ -4,20 +4,62 @@
...
+                ((error line-start (file-name) "(" line "," column "): error " (message) line-end))
+                :modes (typescript-mode))
+
+              (add-to-list 'flycheck-checkers 'typescript)
+              )
+    )
+  )
+
+
+(defun typescript-private/init-tss ()
```
Comment (person808) [inline contrib/!lang/typescript/packages.el]: `typescript/init-tss`

File: contrib/!lang/typescript/packages.el [hunk 5]
```diff
@@ -4,20 +4,62 @@
...
+    :mode (
+           ("\\.ts\\'" . typescript-mode)
+           ))
+  :init (evil-leader/set-key-for-mode 'flycheck-mode
+          "mgg" 'tss-jump-to-definition
+          "mhh" 'tss-popup-help)
+
+  )
+
+(defun typescript-private/post-init-tss ()
```
Comment (person808) [inline contrib/!lang/typescript/packages.el]: `typescript-private/post-init-tss`

Comment (person808) [inline contrib/!lang/typescript/packages.el]: No need for a post-init function here. Just stick add the `:config` section to the `use-package` part of `typescript/init-tss`

File: contrib/!lang/typescript/packages.el [hunk 6]
```diff
@@ -4,20 +4,62 @@
...
   (use-package tss
     :defer t
-    :mode ("\\.ts\\'" . typescript-mode)
-    :init (evil-leader/set-key-for-mode 'typescript-mode
-            "mgg" 'tss-jump-to-definition
-            "mhh" 'tss-popup-help)
-    :config (tss-config-default)))
+    :mode (
+           ("\\.ts\\'" . typescript-mode)
+           ))
```
Comment (person808) [inline contrib/!lang/typescript/packages.el]: Can be simplified to `("\\.ts\\'" . typescript-mode)`

File: contrib/!lang/typescript/packages.el [hunk 7]
```diff
@@ -4,20 +4,62 @@
...
+          "mgg" 'tss-jump-to-definition
+          "mhh" 'tss-popup-help)
+
+  )
+
+(defun typescript-private/post-init-tss ()
+  "Initialize tss"
+  (use-package tss
+    :defer t
+    :config (progn
```
Comment (person808) [inline contrib/!lang/typescript/packages.el]: Put the `(progn` on its own line

File: contrib/!lang/typescript/packages.el [hunk 8]
```diff
@@ -4,20 +4,62 @@
...
 
-(defun typescript/init-tss ()
-  "Initialize my package"
+(setq test-layer-excluded-packages '(flymake))
+
+(defun typescript-private/post-init-flycheck ()
+  "Add hook for flycheck"
+  (use-package flycheck
+    :defer t
+    :config (progn
```
Comment (person808) [inline contrib/!lang/typescript/packages.el]: Put the `(progn` on its own line

File: contrib/!lang/typescript/packages.el [hunk 9]
```diff
@@ -4,20 +4,62 @@
...
     :defer t
-    :mode ("\\.ts\\'" . typescript-mode)
-    :init (evil-leader/set-key-for-mode 'typescript-mode
-            "mgg" 'tss-jump-to-definition
-            "mhh" 'tss-popup-help)
-    :config (tss-config-default)))
+    :mode (
+           ("\\.ts\\'" . typescript-mode)
+           ))
+  :init (evil-leader/set-key-for-mode 'flycheck-mode
```
Comment (person808) [inline contrib/!lang/typescript/packages.el]: The `:init` is part of `use-package`

File: contrib/!lang/typescript/packages.el [hunk 10]
```diff
@@ -4,20 +4,62 @@
...
+                :command ("tsc" "--out" "/dev/null" source-inplace)
+                :error-patterns
+                ((error line-start (file-name) "(" line "," column "): error " (message) line-end))
+                :modes (typescript-mode))
+
+              (add-to-list 'flycheck-checkers 'typescript)
+              )
+    )
+  )
+
```
Comment (person808) [inline contrib/!lang/typescript/packages.el]: Its better to keep trailing parentheses together, don't leave them on their own line

File: contrib/!lang/typescript/packages.el [hunk 11]
```diff
@@ -4,20 +4,50 @@
...
+                            flycheck))
+
+(setq test-layer-excluded-packages '())
+
+(defun typescript/post-init-flycheck ()
+  "Add hook for flycheck"
+  (use-package flycheck
+    :defer t
+    :config (
+             progn
```
Comment (TheBB) [inline contrib/!lang/typescript/packages.el]: What @person808 meant was this.

```lisp
(use-package flycheck
  :defer t
  :config
  (progn
    ...))
```

File: contrib/!lang/typescript/packages.el [hunk 12]
```diff
@@ -4,20 +4,50 @@
...
   (use-package tss
     :defer t
     :mode ("\\.ts\\'" . typescript-mode)
-    :init (evil-leader/set-key-for-mode 'typescript-mode
+    :init (evil-leader/set-key-for-mode 'flycheck-mode
             "mgg" 'tss-jump-to-definition
             "mhh" 'tss-popup-help)
-    :config (tss-config-default)))
+    :config (
+             progn
```
Comment (TheBB) [inline contrib/!lang/typescript/packages.el]: Same here. :-)

File: contrib/!lang/typescript/packages.el (hunk 11 above)
Comment (hamhut1066) [inline contrib/!lang/typescript/packages.el]: right, I thought that it was an odd style

File: contrib/!lang/typescript/packages.el [hunk 13]
```diff
@@ -4,20 +4,50 @@
...
+      (add-to-list 'flycheck-checkers 'typescript))))
 
 (defun typescript/init-tss ()
-  "Initialize my package"
+  "Initialize tss"
   (use-package tss
     :defer t
     :mode ("\\.ts\\'" . typescript-mode)
-    :init (evil-leader/set-key-for-mode 'typescript-mode
+    :init (evil-leader/set-key-for-mode 'flycheck-mode
```
Comment (person808) [inline contrib/!lang/typescript/packages.el]: You probably want to use `evil-leader/set-key-for-mode 'typescript-mode` to keep this from being bound in other filetypes

**Comment by lunaryorn** [comment]
Why not contribute the syntax checker to Flycheck?

**Pull request closed without merging**